### PR TITLE
Range slider

### PR DESCRIPTION
**Pull request by fedderw**

### CODE REVIEW

**Comment by fedderw** [comment]
image

forgive the atrocious blue light filter on the screen shot

File: py/visualize.py [hunk 1]
```diff
@@ -486,48 +475,60 @@ def bubble_chart(responses, poll_ids=None, question_ids=None, xtab1_val="-"):
...
-        annotation_position="bottom left",
-    )
-
     fig.update_layout(
         clickmode="event+select",
         xaxis=dict(
             title=None,  # date self-explanatory
         ),
         legend_title_text=None,  # country self-explanatory
+        autosize=True,
```
Comment (MaxGhenis) [inline py/visualize.py:489]: Could you combine the `update_layout` calls? I think only one `update_traces` call should be necessary too.

File: py/visualize.py [hunk 2]
```diff
@@ -486,48 +475,60 @@ def bubble_chart(responses, poll_ids=None, question_ids=None, xtab1_val="-"):
...
 
     fig.update_traces(
         # Add hover text, which will display instead of custom_data. Custom data is still available to the click event that updates the bar graph
-        # TODO: add plus sign to favorability like in the bar chart
         hovertemplate="<br>".join(
             [
                 "<b>%{customdata[3]}</b>",
                 "%{customdata[2]}",
-                "Net favorability: %{y:+.0f}%",
+                "Net favorability: %{y:+.0f}",
```
Comment (MaxGhenis) [inline py/visualize.py:501]: similarly could you change the y axis title to `Net favorability` (no `%`), if that's not already done? (I didn't see it.)

File: py/visualize.py [hunk 3]
```diff
@@ -486,48 +475,60 @@ def bubble_chart(responses, poll_ids=None, question_ids=None, xtab1_val="-"):
...
+    fig.update_xaxes(rangeslider_visible=True)
+
+    # get the date of the most recent poll, and add one month to it
+    date_range_max = (poll_question.date.max() + timedelta(days=30)).strftime(
+        "%Y-%m-%d"
+    )
+
+    initial_range = ["2016-01-01", date_range_max]
+
+    fig["layout"]["xaxis"].update(range=initial_range)
```
Comment (MaxGhenis) [inline py/visualize.py:521]: i think this could also be part of a big `fig.update_layout` call, as `xaxis_range=initial_range`

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**